app/controllers/vaccinations_controller.rb

```diff
@@ -86,4 +98,13 @@ def set_patient_outcomes
         .order("patients.first_name", "patients.last_name")
         .map { |ps| [ps.patient, ps.outcome] }
   end
+
+  def set_vaccination_record
+    @vaccination_record =
+      @patient
+        .patient_sessions
+        .find_by_session_id(@session.id)
+        .vaccination_records
+        .find_or_initialize_by(recorded_at: nil)
```

Comment:
What do yo think about adding this to `Patient`?

```ruby
  def vaccination_records_for_session(session)
    patient_sessions.find_by_session_id(@session.id).vaccination_records
  end
```

This code would then become:

```suggestion
    @vaccination_record =
      @patient
        .vaccination_records_for_session(@session)
        .find_or_initialize_by(recorded_at: nil)
```